frontend/components/settings/UsernameChange.vue

```diff
@@ -0,0 +1,140 @@
+    validationSchema: yup.object({
+        newUsername: yup
+            .string()
+            .matches(
+                props.usernameRegex,
+                t.value("dashboard.user.settings.username.invalid"),
+            )
+            .required(t.value("form.input.required")),
+    }),
+});
+
+const [newUsername] = defineField("newUsername");
+newUsername.value = props.username;
+
+watch(
+    () => props.visible,
+    (value) => {
+        isVisible.value = value;
+    },
+);
+
+watch(
+    () => props.username,
+    (value) => {
+        newUsername.value = value;
+    },
+);
+
+const onSubmit = handleSubmit(() => {
+    changeUsername();
+});
+
+const close = () => {
+    handleReset();
+    emit("close");
+};
+
+/**
+ * emits to userSettings so that the username can be changed there
+ */
+function changeUsername() {
+    handleReset();
+    emit("changeUsername", newUsername.value);
+}
+</script>
+
+<template>
+    <div>
+        <Sidebar
+            v-model:visible="isVisible"
+            modal
+            position="right"
+            :auto-z-index="true"
+            :draggable="false"
+            class="z-50 mt-auto flex h-[95%] w-full flex-col rounded-t-md bg-background font-nunito dark:bg-background-dark sm:hidden sm:w-4/5 md:rounded-xl lg:-z-10"
+            :pt="{
+                root: {
+                    class: 'font-nunito bg-background dark:bg-background-dark z-10 lg:-z-10 lg:hidden ',
+                },
+                header: {
+                    class: 'flex justify-start pb-2 pl-9 font-nunito bg-background dark:bg-background-dark dark:text-natural-50 rounded-3xl',
+                },
+                title: {
+                    class: 'font-nunito text-4xl font-semibold',
+                },
+                content: {
+                    class: 'font-nunito bg-background dark:bg-background-dark px-0 -ml-2 sm:pr-12 h-full',
+                },
+                footer: { class: 'h-0' },
+                closeButton: {
+                    class: 'justify-start w-full h-full items-center collapse',
+                },
+                mask: {
+                    class: 'sm:collapse bg-natural-50',
+                },
+            }"
+            @hide="close"
+        >
+            <template #header>
+                <button class="-ml-6 flex justify-center pr-4" @click="close">
+                    <span class="pi pi-angle-left text-3xl" />
+                </button>
+                <div class="font-nunito text-4xl font-semibold">
+                    <T key-name="dashboard.user.settings.user.name" />
+                </div>
+            </template>
+            <div class="flex h-full flex-col pl-8">
+                <div
+                    class="-pt-4 overflow-hidden overflow-ellipsis text-sm text-natural-700 dark:text-natural-300"
+                >
+                    <T
+                        key-name="dashboard.user.settings.change.username.description"
```

Comment:
![image](https://github.com/user-attachments/assets/358a68cc-2188-4086-9799-8803a9aaa155)
Please break this text into 3 parts